hnn_core/optimization/general_optimization.py

```diff
@@ -101,6 +108,11 @@ def __init__(
         else:
             self.obj_fun = obj_fun  # user-defined function
             self.obj_fun_name = None
+        # Initial weights for the objective function
+        if initial_params is not None:
+            self.initial_params = initial_params
```

Comment:
Are there any type checks on the format of initial params that get triggered later on?

One good unit test would be to check if there's a mismatch between the keys of `initial_params` and `constraints`